models/Document/Tag/Snippet.php

```diff
@@ -114,71 +115,61 @@ public function frontend()
-            $cacheKey = null;
-            if ($cacheConfig = \Pimcore\Tool\Frontend::isOutputCacheEnabled()) {
+        if (!$this->snippet->isPublished()) {
+            return '';
+        }
 
-                // cleanup params to avoid serializing Element\ElementInterface objects
-                $cacheParams = $params;
-                array_walk($cacheParams, function (&$value, $key) {
-                    if ($value instanceof Model\Element\ElementInterface) {
-                        $value = $value->getId();
-                    }
-                });
+        // apply best matching target group (if any)
+        $targetingConfigurator->configureTargetGroup($this->snippet);
 
-                // TODO is this enough for cache or should we disable caching completely?
-                if ($this->snippet->getUseTargetGroup()) {
-                    $cacheParams['target_group'] = $this->snippet->getUseTargetGroup();
-                }
+        $params = $this->options;
+        $params['document'] = $this->snippet;
 
-                if (Frontend::hasWebpSupport()) {
-                    $cacheParams['webp'] = true;
-                }
+        // check if output-cache is enabled, if so, we're also using the cache here
+        $cacheKey = null;
+        if ($cacheConfig = \Pimcore\Tool\Frontend::isOutputCacheEnabled()) {
 
-                $cacheKey = 'tag_snippet__' . md5(serialize($cacheParams));
-                if ($content = Cache::load($cacheKey)) {
-                    return $content;
+            // cleanup params to avoid serializing Element\ElementInterface objects
+            $cacheParams = $params;
+            array_walk($cacheParams, function (&$value, $key) {
+                if ($value instanceof Model\Element\ElementInterface) {
+                    $value = $value->getId();
                 }
-            }
+            });
 
-            $content = $tagHandler->renderAction(
-                $this->view,
-                $this->snippet->getController(),
-                $this->snippet->getAction(),
-                $this->snippet->getModule(),
-                $params
-            );
-
-            // write contents to the cache, if output-cache is enabled
-            if ($cacheConfig && !DeviceDetector::getInstance()->wasUsed()) {
-                Cache::save($content, $cacheKey, ['output', 'output_inline'], $cacheConfig['lifetime']);
+            // TODO is this enough for cache or should we disable caching completely?
+            if ($this->snippet->getUseTargetGroup()) {
+                $cacheParams['target_group'] = $this->snippet->getUseTargetGroup();
             }
 
-            return $content;
-        } catch (\Exception $e) {
-            Logger::error($e);
+            $cacheParams['webp'] = Frontend::hasWebpSupport();
+            $cacheParams['frontend'] = Tool::isFrontend();
+            $cacheParams['siteRequest'] = Site::isSiteRequest();
```

Comment:
I think we should better use the ID of the site here, then `isFrontend()` shouldn't be necessary.

Reply:
I changed it